curvefs/src/client/s3/client_s3_cache_manager.cpp

```diff
@@ -2017,115 +2023,98 @@ CURVEFS_ERROR DataCache::Flush(uint64_t inodeId, bool force, bool toS3) {
     FSStatusCode ret;
 
     mtx_.lock();
-    if (!force) {
-        if (len_ == chunkSize) {
-            isFlush = true;
-        } else if (now < (createTime_ + flushIntervalSec)) {
-            isFlush = false;
-        }
-    }
-    VLOG(9) << "DataCache::Flush : now:"
-            << now << ",createTime:" << createTime_
+
+    VLOG(9) << "DataCache::Flush : now:" << now << ",createTime:" << createTime_
             << ",flushIntervalSec:" << flushIntervalSec
-            << ",isFlush:" << isFlush << ",chunkPos:" << chunkPos_
-            << ", len:" << len_ << ", inodeId:" << inodeId
-            << ",chunkIndex:" << chunkIndex;
-    if (delete_.load(std::memory_order_acquire)) {
-        VLOG(9) << "datacache is deleted chunkPos:" << chunkPos_
-                << ", len:" << len_ << ", inodeId:" << inodeId
-                << ",chunkIndex:" << chunkIndex;
+            << ",chunkPos:" << chunkPos_ << ", len:" << len_
+            << ", inodeId:" << inodeId << ",chunkIndex:" << chunkIndex;
+
+    tmpLen = len_;
+    blockPos = chunkPos_ % blockSize;
+    blockIndex = chunkPos_ / blockSize;
+    offset = chunkIndex * chunkSize + chunkPos_;
+
+    ret = s3ClientAdaptor_->AllocS3ChunkId(fsId, &chunkId);
+    if (ret != FSStatusCode::OK) {
+        LOG(ERROR) << "alloc s3 chunkid fail. ret:" << ret;
         mtx_.unlock();
-        return CURVEFS_ERROR::NOFLUSH;
-    }
-    if (isFlush) {
-        tmpLen = len_;
-        blockPos = chunkPos_ % blockSize;
-        blockIndex = chunkPos_ / blockSize;
-        offset = chunkIndex * chunkSize + chunkPos_;
-
-        ret = s3ClientAdaptor_->AllocS3ChunkId(fsId, &chunkId);
-        if (ret != FSStatusCode::OK) {
-            LOG(ERROR) << "alloc s3 chunkid fail. ret:" << ret;
-            mtx_.unlock();
-            return CURVEFS_ERROR::INTERNAL;
-        }
-        data = new (std::nothrow) char[len_];
-        if (!data) {
-            LOG(ERROR) << "new data failed.";
-            mtx_.unlock();
-            return CURVEFS_ERROR::INTERNAL;
-        }
-        CopyDataCacheToBuf(0, len_, data);
-        dirty_.store(false, std::memory_order_release);
+        return CURVEFS_ERROR::INTERNAL;
+    }
+    data = new (std::nothrow) char[len_];
+    if (!data) {
+        LOG(ERROR) << "new data failed.";
         mtx_.unlock();
+        return CURVEFS_ERROR::INTERNAL;
+    }
+    CopyDataCacheToBuf(0, len_, data);
+    status_.store(DataCacheStatus::Flush, std::memory_order_release);
+    mtx_.unlock();
```

Comment:
The critical section here is unreasonable.

I think only a few things should be protected by the lock, like line 2032-2035 and line 2049, and others should not.

Reply:
> The critical section here is unreasonable.
> 
> I think only a few things should be protected by the lock, like line 2032-2035 and line 2049, and others should not.
I plan to remove this lock because datacache has been removed from dataWCacheMap_, there is no concurrency